src/ssl.c

```diff
@@ -320,9 +320,59 @@ static int ticket_key_callback_ossl(SSL *ssl, unsigned char *key_name, unsigned
     return ticket_key_callback(key_name, iv, ctx, hctx, enc);
 }
 
-static int encrypt_ticket_key_ptls(ptls_encrypt_ticket_t *self, ptls_t *tls, int is_encrypt, ptls_buffer_t *dst, ptls_iovec_t src)
+static void calculate_quic_tp_tag(uint8_t *tag64, quicly_context_t *ctx)
 {
-    return (is_encrypt ? ptls_openssl_encrypt_ticket : ptls_openssl_decrypt_ticket)(dst, src, ticket_key_callback);
+    ptls_buffer_t buf;
+    uint8_t full_digest[PTLS_SHA256_DIGEST_SIZE];
+
+    /* calculate sha256 hash of remembered tp */
+    ptls_buffer_init(&buf, "", 0);
+    if (quicly_build_session_ticket_auth_data(&buf, ctx) != 0 ||
+        ptls_calc_hash(&ptls_openssl_sha256, full_digest, buf.base, buf.off) != 0)
+        h2o_fatal("failed to calculate sha256 of remembered TPs");
+    ptls_buffer_dispose(&buf);
+
+    /* use the first 64-bit */
+    memcmp(tag64, full_digest, 8);
```

Comment:
This line has no effect? Did you mean `memcpy`?

Reply:
Thank you for catching this.

Now I wonder how my manual testing did not catch the problem, and if we need an automated test to cover this.

Reply:
Nice catch, @hfujita!

Reply:
Fixed in f7df397, added test in 3d4dcd6.